### ISSUE TITLE
Cannot query M2M field via GraphQL

### ISSUE THREAD

**Issue by V1210N** [directus/directus#13987]
### Describe the Bug

Attempting to query an M2M via GraphQL returns the error `Cannot query field "x" on type "y".`

In my case, the collection (x) is "posts" and the M2M field is "categories" (y). The M2M relationship is that of the collection "posts" and the collection "categories", and its field is called "categories" in the posts collection.

The query I'm running is the following:

```
query {
  posts (limit: 10, offset: 0, page: 1) {
    # ... Other fields ommitted for brevity
    
    categories {
      categories_id {
        id
      }
    }
    
    user_created {
      avatar {
        id
      }
      first_name
      last_name
      description
    }
  }
}
```

I believe this query should work as it follows the same structure that worked [in this similar M2M issue](https://github.com/directus/directus/issues/8680).

M2O fields work well, as the `user_created` field is correctly queried, but the `categories` field causes the API to respond with a 400 status code error: `Cannot query field "categories" on type "posts".`

There were no errors when creating the M2M field.

I've tested this multiple times and tried all combinations of:
- Recreating the posts and categories collections before making any changes.
- Calling the "categories" field "categorization" in case there was some naming conflict.
- Making the "categories" field an M2A relationship instead. (Same error)
- Using `posts_categories` (the junction table's name) instead of `categories` as the field name in the query.
- Querying using an administrator account. (Rules out odd permission issues)

### To Reproduce

1. Create a collection called `categories`.
2. Create a collection called `posts`.
3. Create an M2M relationship in `posts` for `categories`.
4. Create an item in `categories`.
5. Create an item in `posts`. Assign the item you created in the previous step to the M2M field.
7. Attempt to run the following query:
```
query {
  posts {
    categories {
      categories_id {
        id
      }
    }
  }
}
```

### Errors Shown

`code`: `GRAPHQL_VALIDATION_EXCEPTION`
`message`:  `Cannot query field "categories" on type "posts".`

### What version of Directus are you using?

9.12.2

### What version of Node.js are you using?

16

### What database are you using?

Postgres 14

### What browser are you using?

(irrelevant)

### How are you deploying Directus?

Digital Ocean

**Comment by azrikahar** [comment]
I can't seem to reproduce this on my end with the provided steps (albeit I added `name` fields to each of them to actually create the records):

https://user-images.githubusercontent.com/42867097/174528804-ce7725ed-b4e0-4c54-b943-046be1eda657.mp4

Not sure if this is cache related on your end, but have you tried restarting Directus (or alternatively use the [clear cache util endpoint](https://docs.directus.io/reference/system/utilities/#clear-the-internal-cache)) before making the graphql query? Just as an additional sanity check here. One possibility might be the schema is cached and was stale during your query 🤔 or perhaps are you running multiple instances of Directus without a shared memory store like redis/memcache? Those are the only few reported cases that I can think of at the moment.

I'll mark this as closed for now, but happy to debug further.

**Comment by V1210N** [comment]
Hey @azrikahar, thanks for checking out my issue.

As you suggested, I've reset the GraphQL cache, yet it still didn't work. Upon further debugging, I've discovered that making requests through API clients works, but using the Directus SDK doesn't.

The issue is in the Directus SDK.

I've just tested my theory by using axios directly and the same query works. I think there is some issue in the default interceptors or functions that run before/after requests that are causing this issue. Though the query string is the same, it fails to be processed via the Directus SDK.

I'm using Directus SDK version 9.12.2, the current latest version.
